src/distributed/cuda/nccl_communicator.cc

```diff
@@ -8,68 +8,140 @@
  * \brief NCCL Communicator.
  */
 
+#include <numeric>
+#include "raf/mpi_communicator.h"
 #include "raf/nccl_communicator.h"
+#include "./nccl_utils.h"
 
 namespace raf {
 namespace distributed {
 namespace communicator {
 
+#define NCCL_UNIQUE_ID_BYTES 128
+
+class NCCLIdSyncHelper {
+ public:
+  NCCLIdSyncHelper() {
+    const char* temp = getenv("RAF_FILE_STORE_PATH");
+    if (temp == nullptr) {
+      LOG(FATAL) << "RAF_FILE_STORE_PATH is no set.";
+    } else {
+      base_path_ = std::string(temp);
+    }
+    this->Append();
+  }
+
+  void Sync(ncclUniqueId* nccl_id, int rank, std::vector<int>& rank_vec) {
+    // rank not in this group, do nothing
+    if (std::find(rank_vec.begin(), rank_vec.end(), rank) == rank_vec.end()) {
+      return;
+    }
+
+    if (rank == rank_vec[0]) {
+      auto vec = std::vector<uint8_t>(reinterpret_cast<uint8_t*>(nccl_id),
+                                      reinterpret_cast<uint8_t*>(nccl_id) + NCCL_UNIQUE_ID_BYTES);
+      fs_vec_.back()->Set(vec);
+    } else {
+      auto vec = fs_vec_.back()->Get();
+      CHECK(vec.size() == NCCL_UNIQUE_ID_BYTES);
+      std::memcpy(nccl_id, vec.data(), vec.size());
+    }
+  }
+
```

Comment:
Could you elaborate a little bit more about the logic of sync and the purpose. It will be better if we have an example here.